File: Chapters/ufcpp2000/csharp/demo/DaifugouWpf/ScriptSample/random.py
from Daifugou import Game

shuffle = lambda lis: [i[1] for i in sorted([[rand.NextDouble(), i] for i in lis], lambda x, y: cmp(x[0], y[0]))]

def play():
	if hand == None or  hand.Length == 0:
		return None

	count = table.Length

	if count == 0:
		count = 1

	if hand.Length < count:
		return None

	for dummy in range(50):
		x = shuffle(hand)
		x = x[0:count]

		if Game.CanPlay(x, table, rank, suit, mode, revolution):
			return x

	return None

result = play()

File: Chapters/ufcpp2000/csharp/demo/DaifugouWpf/ScriptSample/random.cs
using System;
using System.Collections.Generic;
using System.Linq;
using CardGame;
using Daifugou;

/// <summary>
/// ランダムにカードを選んで出すおバカAI。
/// </summary>
public class SampleRandomAi : IArtificialIntelligence
{
    Random rnd = new Random();

    #region IArtificialIntelligence メンバ

    public IEnumerable<Card> Play(IEnumerable<Card> hand, IEnumerable<Card> table, int rank, Suit suit, Mode mode, bool revolution, History history)
    {
        if (hand == null || hand.Count() == 0)
            return null;

        var count = table.Count();

        if (count == 0)
            count = 1;

        if (hand.Count() < count)
            return null;

        for (int i = 0; i < 50; i++)
        {
            var result = hand.OrderBy(x => rnd.Next()).Take(count);
            if(Game.CanPlay(result, table, rank, suit, mode, revolution))
                return result;
        }

        return null;
    }

    #endregion
}
